astropy/utils/tests/test_decorators.py

```diff
@@ -528,6 +530,72 @@ def _get_foo(cls): return 1
     assert B.__dict__['foo'].__doc__ == "The foo."
 
 
+@pytest.fixture
+def fast_thread_switching():
+    """Fixture that reduces thread switching interval.
+
+    This makes it easier to provoke race conditions.
+    """
+    old = sys.getswitchinterval()
```

Comment:
Same question as https://github.com/astropy/astropy/pull/11226#discussion_r555412590 . That is, define this in `setup_module` and `teardown_module` in this case.

Reply:
I don't think it's appropriate to use for the entire module. It will likely slow down the rest of the tests in the module.

I'm also expecting that we'll find more thread safety issues in other modules and write more regression tests, so avoiding duplication of this code seems worthwhile (particularly since I don't yet know what the best value for the magic number is).